spark/src/main/scala/org/apache/spark/sql/delta/stats/StatisticsCollection.scala

```diff
@@ -459,16 +459,22 @@ object StatisticsCollection extends DeltaCommand {
    * @param name The name of the data skipping column for validating data type.
    * @param dataType The data type of the data skipping column.
    * @param columnPaths The column paths of all valid fields.
+   * @param insideStruct Tracks if we are inside a struct, in which case all types are valid
    */
   private def validateDataSkippingType(
       name: String,
       dataType: DataType,
-      columnPaths: ArrayBuffer[String]): Unit = dataType match {
+      columnPaths: ArrayBuffer[String],
+      insideStruct: Boolean = false): Unit = dataType match {
     case s: StructType =>
       s.foreach { field =>
-        validateDataSkippingType(name + "." + field.name, field.dataType, columnPaths)
+        validateDataSkippingType(name + "." + field.name, field.dataType, columnPaths,
+          insideStruct = true)
       }
     case SkippingEligibleDataType(_) => columnPaths.append(name)
+    case d if insideStruct =>
```

Comment:
@Kimahriman 

Could we make this `d` something more descriptive? 

I'm not sure if I follow this part here, why are we appending to the `columnPaths` `name` while logging this message

`"Data skipping is not supported for column $name of type $dataType"`

Will the current description still holds true "The column paths of all valid fields."? If not, I think we should modify it accordingly.

We can put a comment here as well since I'm not really sure what is the full intent of this part here.

Reply:
In this case "valid" means we don't throw an error, not we can actually collect stats on it. We still append to the `columnPaths` because that's purely used to see if the user provided the same column twice. The warning is so users can be aware that stats _won't_ be collected on that column, but we don't throw an error because you didn't specify that column directly, it was inside a struct the user specified.

Reply:
Also `d` isn't used so just changing to `_`

Reply:
We could not append to columnPaths in that case, it doesn't _really_ matter. It's an odd edge case for ensuring the user didn't specify the same struct column multiple times

Reply:
@Kimahriman What does "user-specified struct" mean? I am not sure if I follow "as the user didn't specify them directly", isn't the user that creates the table and the schema?

I think logWarning logs to the Spark Logs right? I'm not sure if it is easily readable in that case as Spark Logs are massive and the messages will be like needle in a haystack.

Could we please update the definition of the function `validateDataSkippingType`, what "valid" mean and `columnPaths` accordingly in that case? The code should be straightforward and easily to understand IMO since the data skipping codebase is already complicated, it took me quite some time to read the function and trying to understand what the intention here, and how it actually works. 

If we have to explain this in a code comment then chances are it is worth making the comment as clear as possible in the codebase as more people will stumble across this function.

Reply:
This function is purely for parsing and validating `DATA_SKIPPING_STATS_COLUMNS`, so "user-specified" in this case means user specified values in `DATA_SKIPPING_STATS_COLUMNS`. I can add to the function comment that this is all the function is for. A user can specify a struct column in this config in which case children fields will automatically be included for stats collection, which is the case this whole PR is about. Any larger confusion about how it works in general or why it works this way would have to go back to the original authors.